tests/contrib/gp/test_models.py

```diff
@@ -26,50 +23,61 @@
 X = torch.tensor([[1., 5., 3.], [4., 3., 7.]])
 y1D = torch.tensor([2., 1.])
 y2D = torch.tensor([[1., 2.], [3., 3.], [1., 4.], [-1., 1.]])
-kernel = RBF(input_dim=3, variance=torch.tensor(3.), lengthscale=torch.tensor(2.))
 noise = torch.tensor(1e-7)
-likelihood = Gaussian(noise)
-
-TEST_CASES = [
-    T(
-        GPRegression,
-        X, y1D, kernel, noise
-    ),
-    T(
-        GPRegression,
-        X, y2D, kernel, noise
-    ),
-    T(
-        SparseGPRegression,
-        X, y1D, kernel, noise
-    ),
-    T(
-        SparseGPRegression,
-        X, y2D, kernel, noise
-    ),
-    T(
-        VariationalGP,
-        X, y1D, kernel, likelihood
-    ),
-    T(
-        VariationalGP,
-        X, y2D, kernel, likelihood
-    ),
-    T(
-        VariationalSparseGP,
-        X, y1D, kernel, likelihood
-    ),
-    T(
-        VariationalSparseGP,
-        X, y2D, kernel, likelihood
-    ),
-]
+
+
+def _kernel():
+    return RBF(input_dim=3, variance=torch.tensor(3.), lengthscale=torch.tensor(2.))
+
+
+def _likelihood():
+    return Gaussian(torch.tensor(1e-7))
+
+
+def _TEST_CASES():
+    TEST_CASES = [
+        T(
+            GPRegression,
+            X, y1D, _kernel(), noise
+        ),
+        T(
+            GPRegression,
+            X, y2D, _kernel(), noise
+        ),
+        T(
+            SparseGPRegression,
+            X, y1D, _kernel(), noise
+        ),
+        T(
+            SparseGPRegression,
+            X, y2D, _kernel(), noise
+        ),
+        T(
+            VariationalGP,
+            X, y1D, _kernel(), _likelihood()
+        ),
+        T(
+            VariationalGP,
+            X, y2D, _kernel(), _likelihood()
+        ),
+        T(
+            VariationalSparseGP,
+            X, y1D, _kernel(), _likelihood()
+        ),
+        T(
+            VariationalSparseGP,
+            X, y2D, _kernel(), _likelihood()
+        ),
+    ]
+
+    return TEST_CASES
```

Comment:
Previously, kernel and likelihood are global variables, so each tests will update their parameters. To prevent that, we need to generate new kernel and likelihood for each test. This is different to before: where learned params are stored in ParamStoreDict and are cleared after each test.